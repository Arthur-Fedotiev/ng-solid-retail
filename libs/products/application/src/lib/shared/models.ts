import { Categories, Colors, Retailers, Tiers } from './constants';

export type CategoriesMap = typeof Categories;
export type Category = typeof Categories[keyof typeof Categories];

export type RetailersMap = typeof Retailers;
export type TiersMap = typeof Tiers;
export type ColorsMap = typeof Colors;
