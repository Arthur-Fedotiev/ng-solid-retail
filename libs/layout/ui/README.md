# layout-ui

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test layout-ui` to execute the unit tests.
