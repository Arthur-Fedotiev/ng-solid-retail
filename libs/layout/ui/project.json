{
  "projectType": "library",
  "sourceRoot": "libs/layout/ui/src",
  "prefix": "sr",
  "targets": {
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/layout/ui"],
      "options": {
        "jestConfig": "libs/layout/ui/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "libs/layout/ui/**/*.ts",
          "libs/layout/ui/**/*.html"
        ]
      }
    }
  },
  "tags": ["scope:layout", "type:ui"]
}
