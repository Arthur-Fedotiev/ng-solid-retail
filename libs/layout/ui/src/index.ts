export * from './lib/layout-ui.module';
