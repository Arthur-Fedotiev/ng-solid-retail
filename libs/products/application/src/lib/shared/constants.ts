import {
  ProductCategory,
  ProductRetailer,
  ProductTier,
  SHOES_AND_CLOTHING_COLORS,
} from '@sr/products/entities';

export const Categories = ProductCategory;
export const Retailers = ProductRetailer;
export const Tiers = ProductTier;
export const Colors = {
  [Categories.Shoes]: SHOES_AND_CLOTHING_COLORS,
  [Categories.Clothing]: SHOES_AND_CLOTHING_COLORS,
} as const;
