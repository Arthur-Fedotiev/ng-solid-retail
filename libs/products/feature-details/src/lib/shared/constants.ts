import { Tiers } from '@sr/products/application';

export const PRODUCT_TIERS = Tiers;
