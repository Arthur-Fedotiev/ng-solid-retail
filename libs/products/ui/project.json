{
  "name": "products-ui",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/products/ui/src",
  "prefix": "sr",
  "targets": {
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/products/ui"],
      "options": {
        "jestConfig": "libs/products/ui/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "libs/products/ui/**/*.ts",
          "libs/products/ui/**/*.html"
        ]
      }
    }
  },
  "tags": ["scope:products", "type:ui"]
}
