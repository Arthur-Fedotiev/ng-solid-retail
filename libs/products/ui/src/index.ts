export * from './lib/products-ui.module';
