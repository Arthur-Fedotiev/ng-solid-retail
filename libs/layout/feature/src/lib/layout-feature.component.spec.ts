import { ComponentFixture, TestBed } from '@angular/core/testing';

import { LayoutFeatureComponent } from './layout-feature.component';
import { RouterTestingModule } from '@angular/router/testing';
import { SharedUiMaterialModule } from '@omnia/shared/ui-material';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { LayoutUiModule } from '@omnia/layout/ui';

describe('LayoutFeatureComponent', () => {
  let component: LayoutFeatureComponent;
  let fixture: ComponentFixture<LayoutFeatureComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule,
        SharedUiMaterialModule,
        NoopAnimationsModule,
        LayoutUiModule,
      ],
      declarations: [LayoutFeatureComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(LayoutFeatureComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
