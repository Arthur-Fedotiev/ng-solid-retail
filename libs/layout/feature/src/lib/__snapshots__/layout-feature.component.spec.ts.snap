// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutFeatureComponent should render 1`] = `
<sr-layout-feature
  __ngContext__={[Function Number]}
>
  <mat-sidenav-container
    class="mat-drawer-container mat-sidenav-container"
    fullscreen=""
  >
    <div
      class="mat-drawer-backdrop"
    />
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav
      class="mat-drawer mat-sidenav ng-trigger ng-trigger-transform mat-drawer-over"
      style="box-shadow:none;visibility:hidden;0:box-shadow;1:visibility;"
      tabindex="-1"
    >
      <div
        cdkscrollable=""
        class="mat-drawer-inner-container"
      >
        <sr-navigation-bar
          class="nav-bar"
        >
          <nav
            class="nav-container"
          >
            <mat-nav-list
              aria-disabled="false"
              class="mat-mdc-nav-list mat-mdc-list-base mdc-list nav-container__list"
              role="navigation"
            >
              <a
                aria-disabled="false"
                class="mat-mdc-list-item mdc-list-item nav-bar__item _mat-animation-noopable mat-mdc-list-item-single-line mdc-list-item--with-one-line"
                href="/products"
                mat-list-item=""
              >
                <span
                  class="mdc-list-item__content"
                >
                  <span
                    class="mat-mdc-list-item-unscoped-content mdc-list-item__primary-text"
                  >
                    <mat-icon
                      aria-hidden="true"
                      class="mat-icon notranslate nav-bar__icon material-icons mat-ligature-font mat-icon-no-color"
                      data-mat-icon-type="font"
                      role="img"
                    >
                      local_grocery_store
                    </mat-icon>
                    <span>
                      Products
                    </span>
                  </span>
                </span>
                <div
                  class="mat-mdc-focus-indicator"
                />
              </a>
              <a
                aria-disabled="false"
                class="mat-mdc-list-item mdc-list-item nav-bar__item _mat-animation-noopable mat-mdc-list-item-single-line mdc-list-item--with-one-line"
                href="/products/create"
                mat-list-item=""
              >
                <span
                  class="mdc-list-item__content"
                >
                  <span
                    class="mat-mdc-list-item-unscoped-content mdc-list-item__primary-text"
                  >
                    <mat-icon
                      aria-hidden="true"
                      class="mat-icon notranslate nav-bar__icon material-icons mat-ligature-font mat-icon-no-color"
                      data-mat-icon-type="font"
                      role="img"
                    >
                      library_add
                    </mat-icon>
                    <span>
                      Create product
                    </span>
                  </span>
                </span>
                <div
                  class="mat-mdc-focus-indicator"
                />
              </a>
            </mat-nav-list>
          </nav>
        </sr-navigation-bar>
      </div>
    </mat-sidenav>
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav-content
      class="mat-drawer-content mat-sidenav-content"
    >
      <mat-drawer-container
        class="mat-drawer-container drawer"
      >
        <div
          class="mat-drawer-backdrop"
        />
        <div
          aria-hidden="true"
          class="cdk-visually-hidden cdk-focus-trap-anchor"
        />
        <mat-drawer
          class="mat-drawer drawer__sidebar ng-trigger ng-trigger-transform mat-drawer-side mat-drawer-opened"
          fxhide.lt-lg=""
          mode="side"
          opened=""
          style="transform:none;visibility:visible;0:transform;1:visibility;"
          tabindex="-1"
        >
          <div
            cdkscrollable=""
            class="mat-drawer-inner-container"
          >
            <button
              class="drawer__trigger mdc-icon-button mat-mdc-icon-button _mat-animation-noopable mat-unthemed mat-mdc-button-base"
              mat-icon-button=""
            >
              <span
                class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
              />
              <mat-icon
                aria-hidden="false"
                aria-label="Menu icon"
                class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
                data-mat-icon-name="menu"
                data-mat-icon-type="font"
                fonticon="menu"
                role="img"
              />
              <span
                class="mat-mdc-focus-indicator"
              />
              <span
                class="mat-ripple mat-mdc-button-ripple"
                matripple=""
              />
              <span
                class="mat-mdc-button-touch-target"
              />
            </button>
          </div>
        </mat-drawer>
        <div
          aria-hidden="true"
          class="cdk-visually-hidden cdk-focus-trap-anchor"
        />
        <mat-drawer-content
          class="mat-drawer-content"
          color="accent"
        >
          <header>
            <sr-profile-toolbar>
              <mat-toolbar
                class="mat-toolbar toolbar mat-toolbar-single-row"
              >
                <button
                  class="drawer__trigger mdc-icon-button mat-mdc-icon-button _mat-animation-noopable mat-unthemed mat-mdc-button-base"
                  fxhide.gt-md=""
                  mat-icon-button=""
                >
                  <span
                    class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
                  />
                  <mat-icon
                    aria-hidden="false"
                    aria-label="Menu icon"
                    class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
                    data-mat-icon-name="menu"
                    data-mat-icon-type="font"
                    fonticon="menu"
                    role="img"
                  />
                  <span
                    class="mat-mdc-focus-indicator"
                  />
                  <span
                    class="mat-ripple mat-mdc-button-ripple"
                    matripple=""
                  />
                  <span
                    class="mat-mdc-button-touch-target"
                  />
                </button>
                <div
                  class="toolbar__profile"
                >
                  <img
                    class="toolbar__avatar"
                    src="https://i.pinimg.com/280x280_RS/a9/7d/6f/a97d6f224a5526c8a49eedc19f0a73c4.jpg"
                  />
                  <span>
                    John Doe
                  </span>
                  <button
                    aria-expanded="false"
                    aria-haspopup="menu"
                    aria-label="Open profile menu"
                    class="mat-mdc-menu-trigger mdc-icon-button mat-mdc-icon-button _mat-animation-noopable mat-unthemed mat-mdc-button-base"
                    mat-icon-button=""
                  >
                    <span
                      class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
                    />
                    <mat-icon
                      aria-hidden="true"
                      class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
                      data-mat-icon-type="font"
                      role="img"
                    >
                      keyboard_arrow_down
                    </mat-icon>
                    <span
                      class="mat-mdc-focus-indicator"
                    />
                    <span
                      class="mat-ripple mat-mdc-button-ripple"
                      matripple=""
                    />
                    <span
                      class="mat-mdc-button-touch-target"
                    />
                  </button>
                  <mat-menu>
                  </mat-menu>
                </div>
              </mat-toolbar>
            </sr-profile-toolbar>
          </header>
          <main
            class="content"
          >
            <router-outlet />
          </main>
        </mat-drawer-content>
      </mat-drawer-container>
    </mat-sidenav-content>
  </mat-sidenav-container>
</sr-layout-feature>
`;
